src/instruments/view/instrumentlistmodel.cpp

```diff
@@ -336,18 +375,171 @@ void InstrumentListModel::setSearchText(const QString& text)
     updateFamilyStateBySearch();
 }
 
-QVariantList InstrumentListModel::instrumentOrderTypes() const
+QVariantList InstrumentListModel::scoreOrders() const
 {
-    // TODO: just for test, needed to implement
+    auto toMap = [](const ScoreOrderInfo& order) {
+        QString name = qApp->translate("instruments", "Order: ") + order.info.name;
+        if (order.customized) {
+            name += qApp->translate("instruments", " (Customized)");
+        }
+        QVariantMap obj;
+        obj[ID_KEY] = order.id;
+        obj[NAME_KEY] = name;
+        obj[CONFIG_KEY] = QVariant::fromValue(order.info);
+        return obj;
+    };
+
     QVariantList result;
-    result << QVariantMap { { ID_KEY, 1 }, { NAME_KEY, qtrc("instruments", "Custom") } };
+    for (const ScoreOrderInfo& order: m_scoreOrders) {
+        result << toMap(order);
+    }
 
     return result;
 }
 
-void InstrumentListModel::selectOrderType(const QString&)
+QVariant InstrumentListModel::selectedScoreOrderIndex() const
 {
-    NOT_IMPLEMENTED;
+    QString id = m_selectedScoreOrderIndex <= 0 ? QString("---") : m_scoreOrders[m_selectedScoreOrderIndex].id;
+    return QVariant(m_selectedScoreOrderIndex);
+}
+
+void InstrumentListModel::setSelectedScoreOrderIndex(const QVariant& index)
+{
+    m_selectedScoreOrderIndex = index.toInt();
```

Comment:
Did you miss emit scoreOrdersChanged here? Because this function is a WRITE method
Q_PROPERTY(QVariant selectedScoreOrderIndex READ selectedScoreOrderIndex WRITE setSelectedScoreOrderIndex NOTIFY scoreOrdersChanged)

Reply:
Indeed, that one was missed.